### PR TITLE
[IMP] Set OCA_ENABLE_CHECKLOG_ODOO by default starting with Odoo 18.0

### PR DESCRIPTION
**Pull request by PabloEForgeFlow**

### CODE REVIEW

**Comment by PabloEForgeFlow** [comment]
@sbidoul I added a question for the setting, it does seem more appropriate.

Regarding version 18.0, now that both the odoo and ocb branches are up, is there something keeping us from adding support here in the template, besides the [oca-ci images](https://github.com/OCA/oca-ci/blob/master/.github/workflows/ci.yaml#L23) not existing yet? Let me know if I can help somehow.

File: copier.yml [hunk 1]
```diff
@@ -157,6 +157,11 @@ include_wkhtmltopdf:
     Do you need to install wkhtmltopdf? Usually only needed if you're going to test PDF
     report generation.
 
+enable_checklog_odoo:
+  type: bool
+  default: "{% if odoo_version < 18.0 -%}no{% else %}yes{% endif %}"
+  help: Do you want warnings to make tests fail?
```
Comment (sbidoul) [inline copier.yml:163]:
```suggestion
  help: Do you want warnings to make tests fail on warnings?
```

**Comment by sbidoul** [comment]
Thanks! Just a minor remark.

> Regarding version 18.0, now that both the odoo and ocb branches are up, is there something keeping us from adding support here in the template, besides the [oca-ci images](https://github.com/OCA/oca-ci/blob/master/.github/workflows/ci.yaml#L23) not existing yet? Let me know if I can help somehow.

We can move forward with that yes. The part that requires most effort is modernizing the linter versions and configs, including pylint-odoo and friends.

File: copier.yml (hunk 1 above)
Comment (pedrobaeza) [reply copier.yml:163]: Shouldn't it be?

```suggestion
  help: Do you want to make tests fail on warnings?
```

Comment (PabloEForgeFlow) [reply copier.yml]: Fixed

**Comment by sbidoul** [comment]
@PabloEForgeFlow this does not always work, because the you added the new env variable in an already conditional section. See https://github.com/OCA/server-env/blob/18.0/.github/workflows/test.yml

**Pull request merged**